### ISSUE TITLE
Invalid Hook call error when using sweetalert in class component

### ISSUE THREAD

**Issue by jslmariano** [kessejones/react-sweetalert2#10]
Hi guys, I'm trying to use it this inside a react class component but i can't seem to make it work.
I'm not sure if i miss something.

```
import React, { Component } from 'react';
import SweetAlert2 from 'react-sweetalert2';

export default class App extends Component{
    constructor(){
        super();

        this.state = {
            swal: {}
        }
    }

    render() {
        return (
            <div>   
                <button onClick={() => {
                    this.setState({
                        swal: {
                            show: true,
                            title: 'Basic Usage',
                            text: 'Hello World'
                        }
                    });
                }}>Alert</button>  
                <SweetAlert2 {...this.state.swal} />
            </div>
        );
    }
}
```

( I actually followed what was in the example )
But i'm getting this error

```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

Can you please help me?

**Comment by kessejones** [comment]
Hi, are you using version 0.4.1?

**Comment by jslmariano** [comment]
Hi @kessejones  yes i am :)

**Comment by kessejones** [comment]
This is very strange. I did some tests here and I didn't have this problem.
Did you create the project with the create-react-app or are you configuring your webpack?

**Comment by jslmariano** [comment]
Hi @kessejones,

VERSIONS:
* react - 16.14.0
* react-scripts - 3.1.1
* react-sweetalert2 - 0.4.1
* sweetalert2 - 10.16.6
* sweetalert2-react-content - 3.3.2

Yeah it was strange, i'm not using a webpack and yes i created it using create-react-app
I already spent a lot of time debugging it though, the problem probably might be rooted deep on some version mismatch in my dependencies.

In the end I used the actual Swal sweetalert and re-created the wrapper based on what you made, and it turned ok. 
Using `withReactContent` for `React.Component` classes


```
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';

...

withReactContent(Swal)
.fire(...props)
.then(result => {
    if(result.isConfirmed)
        onConfirm && onConfirm(result);

    if(result.isDenied)
        onDenied && onDenied(result);

    onResolve && onResolve(result);
  }).catch(error => onError && onError(error));
  
 ...
```

I think the main problem is that I'm using a certain version somewhere that doesn't permit the use of `useEffect` inside a `React.Component`
Though I didn't have the time to explore it more, and I have made it work like this.
